proof_of_concept/asset.py

```diff
@@ -1,21 +1,48 @@
 """Classes for describing assets."""
-from typing import Any, Dict
+from typing import Any, Dict, Union
+
+from proof_of_concept.workflow import Job
+
+
+class Metadata:
```

Comment:
I moved Metadata here due to circular imports (and also makes sense tho have it here as it is not an interface)